.circleci/config.yml

```diff
@@ -0,0 +1,38 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:11.0.2-jdk-node-browsers-legacy
+
+    working_directory: ~/repo
+
+    environment:
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
+
+    steps:
+      - checkout
+      - run: gradle dependencies
+      - run: gradle test
+
+  deploy_integracao:
+    docker:
+      - image: buildpack-deps:trusty
+    steps:
+      - checkout
+      - run:
+          name: Deploy Master to Heroku
+          command: |
+            git push https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_INTEGRATION_APP_NAME.git master 2>&1 >/dev/null | sed "s/$HEROKU_API_KEY/SECRET/g"
```

Comment:
É aqui que a porca torce o rabo, pois precisamos tomar cuidado com duas coisas:

### Configurar a comunicação do Circle CI com o Heroku

As variáveis de ambiente `HEROKU_API_KEY` e `HEROKU_INTEGRATION_APP_NAME` precisam ser configuradas no Circle CI, senão o deploy vai falhar pois o Circle CI não conseguirá se comunicar com o Heroku.

### Processo de deploy duplicado

Hoje, o deploy para o Heroku é feito através da integração direta com o Github. No momento em que esse passo de deploy passar a funcionar via Circle CI, teremos dois deployments acontecendo cada vez que algo for para a master (um via Circle CI e outro via Github), para eliminar essa duplicação, precisamos fazer duas coisas:
 - Garantir que o deploy funciona via Circle CI (em outras palavras, corrigir os itens da primeira observação)
 - Desabilitar a integração direta do Heroku com Github

### Sugestão

1 - Neste momento, configurem o Circle CI para executar somente o `build` da aplicação, sem o deploy. 
2 - Configurem as variáveis de ambiente mencionadas anteriormente
3 - Desabilitem a integração Github/Heroku
4 - Abram uma nova PR configurando o passo de deploy
 


A gente precisa configurar essas duas variáveis de ambiente lá no Circle CI, senão vai dar tudo errado. Se isso ainda não foi feito, pode ser um próximo passo, acho que não é um impeditivo para mergearmos essa PR.